os/arch/arm/src/common/up_restoretask.c

```diff
@@ -0,0 +1,104 @@
+/****************************************************************************
+ *
+ * Copyright 2021 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <sched.h>
+#ifdef CONFIG_SUPPORT_COMMON_BINARY
+#include <stdint.h>
+#endif
+#ifdef CONFIG_TASK_SCHED_HISTORY
+#include <stdbool.h>
+#include <tinyara/debug/sysdbg.h>
+#endif
+#ifdef CONFIG_ARMV8M_TRUSTZONE
+#include <tinyara/tz_context.h>
+#endif
+#ifdef CONFIG_ARM_MPU
+#include <tinyara/mpu.h>
+#include "mpu.h"
+#endif
+#include <tinyara/arch.h>
+
+#include "up_internal.h"
+#include "sched/sched.h"
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+#ifdef CONFIG_SUPPORT_COMMON_BINARY
+extern uint32_t *g_umm_app_id;
+#endif
+
+/************************************************************************************
+ * Name: up_restoretask
+ *
+ * Description:
+ *   Set data needed for context switching from tcb before context switching.
+ *
+ *   This function is called only from the TinyAra scheduling
+ *   logic.
+ *
+ * Inputs:
+ *   tcb: Refers to the tcb which will be scheduled.
+ *     This tcb is at the head of the g_readytorun task lists.
+ *
+ ************************************************************************************/
+void up_restoretask(struct tcb_s *tcb)
```

Comment:
How about `up_restorethreadcontext`?

Reply:
I removed 'context' because the functions with 'context' like up_switchcontext, up_fullcontextrestore switches registers but this just sets tcb data needed for context switching.